Add documentation sections about Qiskit transpile service and AI Transpile passes (#459)
Qiskit/documentation: docs/transpile/ai-transpiler-passes.mdx

])

circuit = EfficientSU2(120, entanglement="circular", reps=1).decompose()

transpiled_circuit = ai_passmanager.run(circuit)
```

Here, the `target` determines which backend to route for, the `optimization_level` (1, 2, or 3) determines the computational effort to spend in the process (higher usually gives better results but takes longer), and the `layout_mode` specifies how to handle the layout selection.
The `layout_mode` includes the following options:
- `keep`: This respects the layout set by the previous transpiler passes (or uses the trivial layout if not set). This is typically only used when it is required to run the circuit on specific qubits of the device. It often produces worse results since it has less room for optimization.
- `improve`: This uses the layout set by the previous transpiler passes as a starting point and improves the existing layout from there. It is useful when you have a good initial guess for the layout; for example,for circuits that are built in a way that approximately follows the device's coupling map. It is also useful if you want to try other specific layout passes combined with the `AIRouting` pass.

> - `optimize`: This is the default mode, and it works best for general circuits where you may not have good layout guesses. This mode ignores previous layout selections and finds a layout on its own.

Reviewer: ```suggestion
- `optimize`: This is the default mode. It works best for general circuits where you might not have good layout guesses. This mode ignores previous layout selections.
```

---
title: AI transpiler passes
description: What are the AI transpiler passes and how to use them
---

# AI transpiler passes
​
The AI-powered transpiler passes are experimental passes that work as a drop-in replacement of "traditional" qiskit passes for some selected transpiling tasks. They often produce better results than existing heuristic algorithms (such as lower depth and CNOT count), but are also much faster than optimization algorithms such as Boolean satisfiability solvers. The AI transpiler passes run on the cloud and are available to IBM Quantum Premium Plan users.

<Admonition type="note">
    This is an experimental feature available only to the IBM Quantum Premium Plan.
    The AI-powered transpiler passes are in alpha release status, subject to change.
</Admonition>

The following passes are currently available:

**Routing passes**
 - `AIRouting`: Layout selection and circuit routing

The following passes will be available in Q1 2024.

**Circuit synthesis passes**
 - `AICliffordSynthesis`: Clifford circuit synthesis
 - `AILinearFunctionSynthesis`: Linear function circuit synthesis
 - `AIPermutationSynthesis`: Permutation circuit synthesis

> ## Install the qiskit-transpiler-service package

Reviewer 2: Do we need to show users how to install the service again if it's already covered in the other page? Instead maybe just say "make sure the qiskit transpiler service is installed, check out the [OTHER PAGE LINK] for details

Reviewer 3: I'll make this change

Author: I do agree, we can do that.

Author: done in https://github.com/Qiskit/documentation/pull/459/commits/eb8de862d646262b21c1935d01b3e63a4b1be1c4

I didn't see your comment @abbycross. Sorry
